File: src/app/model/Algorithm/Algorithm.ts
import { IAlgorithm } from "./IAlgorithm";
import { AlgorithmOutputType } from "../AlgorithmOutputType";
import { Type } from "../../shared/type";

export class Algorithm implements IAlgorithm {
    Name: string;
    UserReadebaleName: string;
    AlgorithmMetaDataParams: Array<[string, Type]>;
    OutputType: AlgorithmOutputType;

    constructor()
    { 
      this.Name = "DummyAlgorithmName";
      this.UserReadebaleName = "Dummy Algorithm User Readebale Name";
      this.AlgorithmMetaDataParams = new Array();
      this.OutputType = AlgorithmOutputType.Text;
    } 
}

File: Server/DsServer/Models/Algorithm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Type = DsServer.Utils.Type;

namespace DsServer.Models
{
    public class Algorithm
    {
        public string Name { get; set; }
        public string UserReadebaleName { get; set; }
        public Tuple<string, Type.TypeEnum>[] AlgorithmMetaDataParams { get; set; }
        public AlgorithmOutputType.AlgorithmOutputTypeEnum OutputType { get; set; }
    }
}
